apps/webapp/vite.config.ts

```diff
@@ -1,8 +1,19 @@
-import { defineConfig } from 'vite';
+import { defineConfig, splitVendorChunkPlugin } from 'vite';
 import react from '@vitejs/plugin-react-swc';
 
 // https://vitejs.dev/config/
 export default defineConfig({
   base: './',
-  plugins: [react()],
+  plugins: [react(), splitVendorChunkPlugin()],
+  build: {
+    rollupOptions: {
+      output: {
+        manualChunks: (id: string) => {
+          if (id.includes('@buf')) return 'protobuf';
+          else if (id.includes('node_modules')) return 'vendor';
+          return;
```

Comment:
What does this do? Can you add a code comment?

Reply:
this is the chunking config for vite. protobuf is a nice conceptual unit, so i put it in its own chunk. vendor chunk is just anything from a package.

i guess line 13 is unnecessary, it is achieved by the vendor chunk plugin above.

https://vitejs.dev/guide/build.html#chunking-strategy

Reply:
I think this is fine for now. But likely the best chunking possible is by page. This is something meta-frameworks do automatically for us and another reason to explore them.